awswrangler/s3/_read_parquet.py

```diff
@@ -1,86 +1,94 @@
+FULL_READ_S3_BLOCK_SIZE = 20_971_520  # 20 MB (20 * 2**20)
+METADATA_READ_S3_BLOCK_SIZE = 131_072  # 128 KB (128 * 2**10)
+
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
+def _ensure_locations_are_valid(paths: Iterable[str]) -> Iterator[str]:
+    for path in paths:
+        suffix: str = path.rpartition("/")[2]
+        # If the suffix looks like a partition,
+        if (suffix != "") and (suffix.count("=") == 1):
+            # the path should end in a '/' character.
+            path = f"{path}/"
+        yield path
+
+
 def _pyarrow_parquet_file_wrapper(
-    source: Any, read_dictionary: Optional[List[str]] = None, coerce_int96_timestamp_unit: Optional[str] = None
+    source: Any, coerce_int96_timestamp_unit: Optional[str] = None
 ) -> pyarrow.parquet.ParquetFile:
     try:
-        try:
-            return pyarrow.parquet.ParquetFile(
-                source=source, read_dictionary=read_dictionary, coerce_int96_timestamp_unit=coerce_int96_timestamp_unit
-            )
-        except TypeError as ex:
-            if "got an unexpected keyword argument" in str(ex):
-                _logger.warning("coerce_int96_timestamp_unit is not supported in pyarrow 2 and below")
-                return pyarrow.parquet.ParquetFile(source=source, read_dictionary=read_dictionary)
-            raise
+        return pyarrow.parquet.ParquetFile(source=source, coerce_int96_timestamp_unit=coerce_int96_timestamp_unit)
     except pyarrow.ArrowInvalid as ex:
         if str(ex) == "Parquet file size is 0 bytes":
-            _logger.warning("Ignoring empty file...xx")
+            _logger.warning("Ignoring empty file...")
             return None
         raise
 
 
+@ray_remote
 def _read_parquet_metadata_file(
-    path: str,
     boto3_session: boto3.Session,
+    path: str,
     s3_additional_kwargs: Optional[Dict[str, str]],
     use_threads: Union[bool, int],
     version_id: Optional[str] = None,
-    ignore_null: bool = False,
-    pyarrow_additional_kwargs: Optional[Dict[str, Any]] = None,
-) -> Optional[Dict[str, str]]:
-    pyarrow_args = _set_default_pyarrow_additional_kwargs(pyarrow_additional_kwargs)
+    coerce_int96_timestamp_unit: Optional[str] = None,
+) -> pa.schema:
     with open_s3_object(
         path=path,
         mode="rb",
         version_id=version_id,
         use_threads=use_threads,
-        s3_block_size=131_072,  # 128 KB (128 * 2**10)
+        s3_block_size=METADATA_READ_S3_BLOCK_SIZE,
         s3_additional_kwargs=s3_additional_kwargs,
         boto3_session=boto3_session,
     ) as f:
         pq_file: Optional[pyarrow.parquet.ParquetFile] = _pyarrow_parquet_file_wrapper(
-            source=f, coerce_int96_timestamp_unit=pyarrow_args["coerce_int96_timestamp_unit"]
+            source=f, coerce_int96_timestamp_unit=coerce_int96_timestamp_unit
         )
-        if pq_file is None:
-            return None
-        return _data_types.athena_types_from_pyarrow_schema(
```

Comment:
Could you explain why no longer type cast here?

Reply:
The type casting has moved [here](https://github.com/awslabs/aws-data-wrangler/blob/feat-3.0/distributed-s3-read-parquet/awswrangler/s3/_read_parquet.py#L198). I believe this is better for a number of reasons:
1. Type casting is only done once when required, not when reading the metadata of each file
2. I think `_read_parquet_metadata_file` is doing too many things. It should just return the schema of a file, no conversion. Indeed, I had to call the method more than once, and in some cases I only needed the schema. If it returns a dict of types, I would have to reconvert back to a schema...